feat: Add Core Web Vitals path breakdown

rafaeelaudibert opened the pull request:
This query returns a list of paths and whether their performance is good, needs improvements, or poor. We then display those in the UI in a custom table.

![image](https://github.com/user-attachments/assets/d956c13d-d094-408b-a75e-a67f7ca76547)


## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Unit tests for backend
Manual tests for UI

rafaeelaudibert committed:
feat: Add CoreWebVitalsPathBreakdownQueryRunner

This query returns a list of paths and whether their performance is good, needs improvements or poor. Follow-up commit will add a UI for this new query. There's a dummy JSON.stringify in the UI at the moment.

rafaeelaudibert committed:
feat: Add Core Web Vitals Path Breakdown UI

Heavily inspired on Vercel's UI, I think they nailed this, and we'll
start with a similar interface

rafaeelaudibert committed:
fix: Use sentence case

Don't agree 100%, but this is our standard

rafaeelaudibert committed:
fix: Remove backend type errors

rafaeelaudibert committed:
refactor: Fix missing `CoreWebVitalsPathBreakdownQuery` on `AnyDataNode` type

rafaeelaudibert committed:
fix: More lint errors

rafaeelaudibert committed:
fix: Regenerate schema

robbie-c commented on frontend/src/queries/schema/schema-general.ts:
```diff
@@ -1563,6 +1572,26 @@ export interface CoreWebVitalsItem {
 export type CoreWebVitalsQueryResponse = AnalyticsQueryResponseBase<CoreWebVitalsItem[]>
 export type CachedCoreWebVitalsQueryResponse = CachedQueryResponse<CoreWebVitalsQueryResponse>
 
+export interface CoreWebVitalsPathBreakdownQuery extends WebAnalyticsQueryBase<WebGoalsQueryResponse> {
```
```suggestion
export interface CoreWebVitalsPathBreakdownQuery extends WebAnalyticsQueryBase<CoreWebVitalsPathBreakdownResult> {
```

robbie-c commented on posthog/hogql_queries/web_analytics/core_web_vitals_path_breakdown.py:
```diff
@@ -0,0 +1,146 @@
+
+    def to_query(self):
+        return parse_select(
+            "SELECT * FROM {good_query} UNION ALL {needs_improvements_query} UNION ALL {poor_query}",
```
Not 100% sure, but I don't think this is going to be optimised by clickhouse at all. I think it'll do something like 3 queries and stitch the results together. There might be a more performant way of structuring this.

(don't let this stop you from merging, it's fine to make it work, and make it fast, in 2 separate steps)

rafaeelaudibert replied:
Yeah, I'm pretty sure that's 3 queries, unfortunately. I've tried a new approach in 9852a28523df02ea194d6dba721ef04969acaadf

robbie-c reviewed (approved):
Small fix and a a thought about query speed for the future

rafaeelaudibert committed:
test: Improve limit 20 test

This will allow us to tweak the query but guarantee that we're still only returning at most 20 per level

rafaeelaudibert committed:
refactor: Optimize UNION ALL query for Core Web Vitals

UNION ALL is very inneficient becaue it's simply 3 separate queries
processed in parallel. We can do better, because those 3 queries were
very similar. We're now grabbing all of the rows and then using LIMIT BY
to get at most 20 entries per band

rafaeelaudibert committed:
fix: Schema incorrectly typed
